Document `LOAD ON ALL DATA` privilege (#737)
neo4j/docs-cypher: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc

| Introduction of a xref::functions/scalar.adoc#functions-valueType[valueType()] function.
This function returns a `STRING` representation of the most precise value xref::values-and-types/property-structural-constructed.adoc#types-synonyms[type] that the given expression evaluates to.

a|
label:functionality[]
label:new[]

New privilege:
[source, cypher, role=noheader]
----
GRANT LOAD ON ALL DATA TO `role`
----

| New privilege that controls a user's ability to load data.

> In difference to existing privileges these are not applied to DBMS, database or graph.

Reviewer: ```suggestion
In difference to other privileges these are not specific to DBMS, database or graph. They apply system wide instead.
```

- `existing` vs `other` -> LOAD ON ALL DATA is different to "other" future privileges as well, not only old `existing` privileges
- `not applied to` vs `not specific to` -> `not applied to` could be misunderstood as "does not work with" where we want to say "works regardless of these"

Author: >  is different to "other" future privileges as well

not other future LOAD privileges and who knows what other privileges we'll add but I see your point

Author: technically DBMS privileges are also system wide though... both the load and dbms privileges will apply on the same level (compare with EXECUTE for example).

What I'm after with the sentence is to say you can't have `LOAD ALL DATA ON DBMS`, you don't have the same privilege target. I'm not sure how to describe it reasonably

Reviewer: I assumed you where highlighting it because it affects the scope. I'm not sure if the privilege target on it's own is relevant here. If we don't care about the scope, all that is left is syntax, and that doesn't need to be explained in the changelog?

Author: target/scope sure, I kinda wanted to point out that this is the first privilege that doesn't follow the same pattern of database/graph/dbms that all our current privileges have. I probably should have a paragraph about this in the load chapter as well though, with more details/explanation.

(would have made sense to me to have this be another dbms privilege like execute (something like `GRANT LOAD ON DBMS ALL DATA TO role` or `GRANT LOAD ALL DATA ON DBMS TO role`) but that wasn't what was decided on 🤷)

> In difference to other privileges these are not granted, denied, or revoked on DBMS, database or graph but instead simply on `ALL DATA`.

Reviewer 2: ```suggestion
Unlike other privileges, these are not granted, denied, or revoked on graphs, databases, or the DBMS, but instead on `ALL DATA`.
```
